src/test/java/com/Kuba2412/medicalclinic/controller/DoctorControllerTest.java

```diff
@@ -0,0 +1,111 @@
+import static org.mockito.ArgumentMatchers.anyLong;
+import static org.mockito.Mockito.when;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.*;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.*;
+
+@SpringBootTest
+@AutoConfigureMockMvc
+public class DoctorControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private DoctorService doctorService;
+
+    private DoctorDTO doctorDTO;
+    private SimpleDoctorDTO simpleDoctorDTO;
+    private Institution institution;
+
+    @BeforeEach
+    void setup() {
+        doctorDTO = new DoctorDTO();
+        doctorDTO.setFirstName("Kuba");
+        doctorDTO.setLastName("Ppp");
+        doctorDTO.setEmail("kuba123@gmail.com");
+
+        simpleDoctorDTO = new SimpleDoctorDTO();
+        simpleDoctorDTO.setFirstName("Kuba");
+        simpleDoctorDTO.setLastName("Ppp");
+
+        institution = new Institution();
+        institution.setName("Szpital 1");
+    }
+
+    @Test
+    void createDoctor_ValidInput_DoctorCreated() throws Exception {
+        mockMvc.perform(post("/doctors")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(doctorDTO)))
+                .andExpect(status().isCreated())
+                .andExpect(content().string("Doctor created successfully."));
+    }
+
+    @Test
+    void getAllDoctors_DoctorsExist_DoctorsReturned() throws Exception {
+        List<SimpleDoctorDTO> doctors = Arrays.asList(simpleDoctorDTO);
+        when(doctorService.getAllDoctors(any())).thenReturn(doctors);
+
+        mockMvc.perform(get("/doctors")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size()").value(doctors.size()));
+    }
+
+    @Test
+    void getAllSimpleDoctors_DoctorsExist_DoctorsReturned() throws Exception {
+        List<SimpleDoctorDTO> doctors = Arrays.asList(simpleDoctorDTO);
+        when(doctorService.getAllSimpleDoctors(any())).thenReturn(doctors);
+
+        mockMvc.perform(get("/doctors/simple")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size()").value(doctors.size()));
+    }
+
+    @Test
+    void getAssignedInstitutionsForDoctor_DoctorExists_InstitutionsReturned() throws Exception {
+        List<Institution> institutions = Arrays.asList(institution);
+        when(doctorService.getAssignedInstitutionsForDoctor(anyLong())).thenReturn(institutions);
+
+        mockMvc.perform(get("/doctors/1/institutions")
+                        .contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.size()").value(institutions.size()));
+    }
```

Comment:
Tutaj też można by było sprawdzić czy np nazwa instytucji to ta która została przekazana